---
title: "Benchmarking pipeline resources - per rule - research server"
author: "Leah Kemp"
date: "9/15/2020"
output: rmarkdown::github_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = FALSE, warning = FALSE, error = FALSE, message = FALSE)
```

## Context

This document aims to plot and analyse the results of resource benchmarking tests for our genomic pipelines on a single research server (Wintermute). See related docs here: [resource benchmarking per rule research server](../resource_benchmarking_per_rule_research_server.md)

```{r}
# Data preparation
library(vcfR)
library(ggplot2)
library(dplyr)
library(gridExtra)

# Read in data
data <- read.csv("resource_benchmarking_merged.csv")

# Make minutes and hours columns calculated from seconds column
data <- data %>% dplyr::mutate(minutes = s/60)
data <- data %>% dplyr::mutate(hours = minutes/60)

# Make max_uss GB columns calculated from max_uss column
data <- data %>% dplyr::mutate(max_uss_gb = max_uss*0.00104858)

# Extract data by rule
gatk_HaplotypeCaller_single_data <- data %>%
  dplyr::filter(rule == "gatk_HaplotypeCaller_single")
gatk_HaplotypeCaller_cohort_data <- data %>%
  dplyr::filter(rule == "gatk_HaplotypeCaller_cohort")
gatk_BaseRecalibrator_data <- data %>%
  dplyr::filter(rule == "gatk_BaseRecalibrator")
bwa_mem_data <- data %>%
  dplyr::filter(rule == "bwa_mem")
gatk_ApplyBQSR_data <- data %>%
  dplyr::filter(rule == "gatk_ApplyBQSR")
trim_galore_pe_data <- data %>%
  dplyr::filter(rule == "trim_galore_pe")
gatk_MarkDuplicates_data <- data %>%
  dplyr::filter(rule == "gatk_MarkDuplicates")
fastqc_data <- data %>%
  dplyr::filter(rule == "fastqc")
gatk_CNNScoreVariants_data <- data %>%
  dplyr::filter(rule == "gatk_CNNScoreVariants")
gatk_FilterVariantTranches_data <- data %>%
  dplyr::filter(rule == "gatk_FilterVariantTranches")
SnpSift_dbnsfp_data <- data %>%
  dplyr::filter(rule == "SnpSift_dbnsfp")
genmod_annotate_CADD_data <- data %>%
  dplyr::filter(rule == "genmod_annotate_CADD")
SnpSift_annotate_dbSNP_data <- data %>%
  dplyr::filter(rule == "SnpSift_annotate_dbSNP")
bcftools_view_multiallelicsites_data <- data %>%
  dplyr::filter(rule == "bcftools_view_multiallelicsites")
genmod_score_data <- data %>%
  dplyr::filter(rule == "genmod_score")
gatk_CombineGVCFs_data <- data %>%
  dplyr::filter(rule == "gatk_CombineGVCFs")
gatk_GenotypeGVCFs_data <- data %>%
  dplyr::filter(rule == "gatk_GenotypeGVCFs")
gatk_VariantRecalibrator_indel_data <- data %>%
  dplyr::filter(rule == "gatk_VariantRecalibrator_indel")
gatk_VariantRecalibrator_snp_data <- data %>%
  dplyr::filter(rule == "gatk4_VariantRecalibrator_snp")
gatk_VQSR_indel_data <- data %>%
  dplyr::filter(rule == "gatk_VQSR_indel")
gatk_VQSR_snp_data <- data %>%
  dplyr::filter(rule == "gatk_VQSR_snp")
vep_data <- data %>%
  dplyr::filter(rule == "vep")
gatk_CalculateGenotypePosteriors_data <- data %>%
  dplyr::filter(rule == "gatk_CalculateGenotypePosteriors")
gatk_VariantAnnotator_PossibleDeNovo_data <- data %>%
  dplyr::filter(rule == "gatk_VariantAnnotator_PossibleDeNovo")
genmod_models_data <- data %>%
  dplyr::filter(rule == "genmod_models")
SnpSift_filter_proband_data <- data %>%
  dplyr::filter(rule == "SnpSift_filter_proband")
```

## Threading

### Find optimal threads for each rule (diminishing return in speed)

```{r}
xlab = "Threads"
ylab = "Minutes"
secondaxis = "Hours"

fastqc_plot <- ggplot2::ggplot(data = fastqc_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "fastqc", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,10), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

trim_galore_pe_plot <- ggplot2::ggplot(data = trim_galore_pe_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "trim_galore_pe", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,100), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 8, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

bwa_mem_plot <- ggplot2::ggplot(data = bwa_mem_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "bwa_mem", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,500), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 4, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_MarkDuplicates_plot <- ggplot2::ggplot(data = gatk_MarkDuplicates_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_MarkDuplicates", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,30), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_BaseRecalibrator_plot <- ggplot2::ggplot(data = gatk_BaseRecalibrator_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_BaseRecalibrator", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,60), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_ApplyBQSR_plot <- ggplot2::ggplot(data = gatk_ApplyBQSR_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_ApplyBQSR", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,60), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_HaplotypeCaller_single_plot <- ggplot2::ggplot(data = gatk_HaplotypeCaller_single_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_HaplotypeCaller_single", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,400), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_HaplotypeCaller_cohort_plot <- ggplot2::ggplot(data = gatk_HaplotypeCaller_cohort_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_HaplotypeCaller_cohort", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,400), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_CNNScoreVariants_plot <- ggplot2::ggplot(data = gatk_CNNScoreVariants_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_CNNScoreVariants", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,1000), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 32, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_FilterVariantTranches_plot <- ggplot2::ggplot(data = gatk_FilterVariantTranches_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_FilterVariantTranches", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,1), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

SnpSift_dbnsfp_plot <- ggplot2::ggplot(data = SnpSift_dbnsfp_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "SnpSift_dbnsfp", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,400), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

genmod_annotate_CADD_plot <- ggplot2::ggplot(data = genmod_annotate_CADD_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "genmod_annotate_CADD", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,100), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

SnpSift_annotate_dbSNP_plot <- ggplot2::ggplot(data = SnpSift_annotate_dbSNP_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "SnpSift_annotate_dbSNP", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,3), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

bcftools_view_multiallelicsites_plot <- ggplot2::ggplot(data = bcftools_view_multiallelicsites_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "bcftools_view_multiallelicsites", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,1), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

genmod_score_plot <- ggplot2::ggplot(data = genmod_score_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "genmod_score", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,2), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_CombineGVCFs_plot <- ggplot2::ggplot(data = gatk_CombineGVCFs_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_CombineGVCFs", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,30), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_GenotypeGVCFs_plot <- ggplot2::ggplot(data = gatk_GenotypeGVCFs_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_GenotypeGVCFs", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,30), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_VariantRecalibrator_indel_plot <- ggplot2::ggplot(data = gatk_VariantRecalibrator_indel_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_VariantRecalibrator_indel", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,30), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_VariantRecalibrator_snp_plot <- ggplot2::ggplot(data = gatk_VariantRecalibrator_snp_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_VariantRecalibrator_snp", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,30), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_ApplyVQSR_indel_plot <- ggplot2::ggplot(data = gatk_VQSR_indel_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_ApplyVQSR_indel", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,1), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_ApplyVQSR_snp_plot <- ggplot2::ggplot(data = gatk_VQSR_snp_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_ApplyVQSR_snp", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,1), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

vep_plot <- ggplot2::ggplot(data = vep_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "vep", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,30), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_CalculateGenotypePosteriors_plot <- ggplot2::ggplot(data = gatk_CalculateGenotypePosteriors_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_CalculateGenotypePosteriors", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,3), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_VariantAnnotator_PossibleDeNovo_plot <- ggplot2::ggplot(data = gatk_VariantAnnotator_PossibleDeNovo_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_VariantAnnotator_PossibleDeNovo", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,3), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

genmod_models_plot <- ggplot2::ggplot(data = genmod_models_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "genmod_models", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,20), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

SnpSift_filter_proband_plot <- ggplot2::ggplot(data = SnpSift_filter_proband_data, aes(x = threads_per_rule, y = minutes, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "SnpSift_filter_proband", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,3), 
    sec.axis = sec_axis(~ ./60, name = secondaxis)) +
  geom_vline(xintercept = 1, linetype = "dashed") + 
  theme(legend.position = "none", plot.title = element_text(size=10))

```

#### human_genomics_pipeline

```{r}
# Plot plots
grid.arrange(fastqc_plot,
             trim_galore_pe_plot,
             bwa_mem_plot,
             gatk_MarkDuplicates_plot,
             ncol = 2, nrow = 2)
             
grid.arrange(gatk_BaseRecalibrator_plot,
             gatk_ApplyBQSR_plot,
             gatk_HaplotypeCaller_single_plot,
             gatk_HaplotypeCaller_cohort_plot,
             ncol = 2, nrow = 2)    

grid.arrange(gatk_CombineGVCFs_plot,
             gatk_GenotypeGVCFs_plot,
             ncol = 2, nrow = 2)
```

#### vcf_annotation_pipeline

```{r}
# Plot plots
 

grid.arrange(gatk_CNNScoreVariants_plot,
             gatk_FilterVariantTranches_plot,
             gatk_VariantRecalibrator_indel_plot,
             gatk_VariantRecalibrator_snp_plot,
             ncol = 2, nrow = 2)

grid.arrange(gatk_ApplyVQSR_indel_plot,
             gatk_ApplyVQSR_snp_plot,
             SnpSift_dbnsfp_plot,
             vep_plot,
             ncol = 2, nrow = 2)

grid.arrange(genmod_annotate_CADD_plot,
             SnpSift_annotate_dbSNP_plot,
             gatk_CalculateGenotypePosteriors_plot,
             gatk_VariantAnnotator_PossibleDeNovo_plot,
             ncol = 2, nrow = 2)

grid.arrange(genmod_models_plot,
             bcftools_view_multiallelicsites_plot,
             SnpSift_filter_proband_plot,
             genmod_score_plot,
             ncol = 2, nrow = 2)
             
```

### Find maximum memory usage for each rule (max USS)

```{r}
xlab = "Threads"
ylab = "Mebibyte"
secondaxis = "Gigabyte"

fastqc_memory_plot <- ggplot2::ggplot(data = fastqc_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "fastqc", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

trim_galore_pe_memory_plot <- ggplot2::ggplot(data = trim_galore_pe_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "trim_galore_pe", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

bwa_mem_memory_plot <- ggplot2::ggplot(data = bwa_mem_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "bwa_mem", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_MarkDuplicates_memory_plot <- ggplot2::ggplot(data = gatk_MarkDuplicates_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_MarkDuplicates", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_BaseRecalibrator_memory_plot <- ggplot2::ggplot(data = gatk_BaseRecalibrator_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_BaseRecalibrator", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_ApplyBQSR_memory_plot <- ggplot2::ggplot(data = gatk_ApplyBQSR_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_ApplyBQSR", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_HaplotypeCaller_single_memory_plot <- ggplot2::ggplot(data = gatk_HaplotypeCaller_single_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_HaplotypeCaller_single", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_HaplotypeCaller_cohort_memory_plot <- ggplot2::ggplot(data = gatk_HaplotypeCaller_cohort_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_HaplotypeCaller_cohort", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_CNNScoreVariants_memory_plot <- ggplot2::ggplot(data = gatk_CNNScoreVariants_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_CNNScoreVariants", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_FilterVariantTranches_memory_plot <- ggplot2::ggplot(data = gatk_FilterVariantTranches_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_FilterVariantTranches", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

SnpSift_dbnsfp_memory_plot <- ggplot2::ggplot(data = SnpSift_dbnsfp_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "SnpSift_dbnsfp", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

genmod_annotate_CADD_memory_plot <- ggplot2::ggplot(data = genmod_annotate_CADD_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "genmod_annotate_CADD", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

SnpSift_annotate_dbSNP_memory_plot <- ggplot2::ggplot(data = SnpSift_annotate_dbSNP_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "SnpSift_annotate_dbSNP", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

bcftools_view_multiallelicsites_memory_plot <- ggplot2::ggplot(data = bcftools_view_multiallelicsites_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "bcftools_view_multiallelicsites", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

genmod_score_memory_plot <- ggplot2::ggplot(data = genmod_score_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "genmod_score", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_CombineGVCFs_memory_plot <- ggplot2::ggplot(data = gatk_CombineGVCFs_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_CombineGVCFs", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2,4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_GenotypeGVCFs_memory_plot <- ggplot2::ggplot(data = gatk_GenotypeGVCFs_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_GenotypeGVCFs", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_VariantRecalibrator_indel_memory_plot <- ggplot2::ggplot(data = gatk_VariantRecalibrator_indel_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_VariantRecalibrator_indel", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_VariantRecalibrator_snp_memory_plot <- ggplot2::ggplot(data = gatk_VariantRecalibrator_snp_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_VariantRecalibrator_snp", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_VQSR_indel_memory_plot <- ggplot2::ggplot(data = gatk_VQSR_indel_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_VQSR_indel", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_VQSR_snp_memory_plot <- ggplot2::ggplot(data = gatk_VQSR_snp_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_VQSR_snp", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

vep_memory_plot <- ggplot2::ggplot(data = vep_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "vep", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_CalculateGenotypePosteriors_memory_plot <- ggplot2::ggplot(data = gatk_CalculateGenotypePosteriors_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_CalculateGenotypePosteriors", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

gatk_VariantAnnotator_PossibleDeNovo_memory_plot <- ggplot2::ggplot(data = gatk_VariantAnnotator_PossibleDeNovo_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "gatk_VariantAnnotator_PossibleDeNovo", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

genmod_models_memory_plot <- ggplot2::ggplot(data = genmod_models_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "genmod_models", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

SnpSift_filter_proband_memory_plot <- ggplot2::ggplot(data = SnpSift_filter_proband_data, aes(x = threads_per_rule, y = max_uss, group = filepath)) +
  geom_point() +
  geom_line() +
  labs(title = "SnpSift_filter_proband", x = xlab, y = ylab) +
  scale_x_continuous(breaks=c(1, 2, 4, 8, 16, 32), limits=c(1, 32), trans='log10') +
  scale_y_continuous(limits=c(0,50000), 
    sec.axis = sec_axis(~ .*0.00104858, name = secondaxis)) +
  theme(legend.position = "none", plot.title = element_text(size=10))

```

#### human_genomics_pipeline

```{r}
# Plot plots
grid.arrange(fastqc_memory_plot,
             trim_galore_pe_memory_plot,
             bwa_mem_memory_plot,
             gatk_MarkDuplicates_memory_plot,
             ncol = 2, nrow = 2)
             
grid.arrange(gatk_BaseRecalibrator_memory_plot,
             gatk_ApplyBQSR_memory_plot,
             gatk_HaplotypeCaller_single_memory_plot,
             gatk_HaplotypeCaller_cohort_memory_plot,
             ncol = 2, nrow = 2)    

grid.arrange(gatk_CombineGVCFs_memory_plot,
             gatk_GenotypeGVCFs_memory_plot,
             ncol = 2, nrow = 2)
```

#### vcf_annotation_pipeline

```{r}
# Plot plots
 

grid.arrange(gatk_CNNScoreVariants_memory_plot,
             gatk_FilterVariantTranches_memory_plot,
             gatk_VariantRecalibrator_indel_memory_plot,
             gatk_VariantRecalibrator_snp_memory_plot,
             ncol = 2, nrow = 2)

grid.arrange(gatk_VQSR_indel_memory_plot,
             gatk_VQSR_snp_memory_plot,
             SnpSift_dbnsfp_memory_plot,
             vep_memory_plot,
             ncol = 2, nrow = 2)

grid.arrange(genmod_annotate_CADD_memory_plot,
             SnpSift_annotate_dbSNP_memory_plot,
             gatk_CalculateGenotypePosteriors_memory_plot,
             gatk_VariantAnnotator_PossibleDeNovo_memory_plot,
             ncol = 2, nrow = 2)

grid.arrange(genmod_models_memory_plot,
             bcftools_view_multiallelicsites_memory_plot,
             SnpSift_filter_proband_memory_plot,
             genmod_score_memory_plot,
             ncol = 2, nrow = 2)
             
```




